main/src/hadd.cxx

```diff
@@ -1,39 +1,51 @@
 /**
   \file hadd.cxx
-  \brief This program will add histograms (see note) and Trees from a list of root files and write them to a target root file.
-  The target file is newly created and must not be
-  identical to one of the source files.
+  \brief This program will merge compatible ROOT objects, such as histograms and Trees,
+         from a list of root files and write them to a target root file.
+         In order for a ROOT object to be mergeable, it must implement TObject's Merge() function.
```

Comment:
It is probably fine as it is but if I recall correctly the object's class "just" need to implement `Merge` independently of the inheritance from `TObject`.

Reply:
As far as I see in TFileMerger, RNTuple is the only supported non-TObject. But I might be looking at the wrong thing

Reply:
See https://github.com/root-project/root/blob/0cfe54439fc766ab9008a228d664852c7fc634a5/io/io/src/TFileMerger.cxx#L587 and https://github.com/root-project/root/blob/0cfe54439fc766ab9008a228d664852c7fc634a5/io/io/src/TFileMerger.cxx#379

Reply:
Ah, my bad, I totally missed the `!` of `!cl->IsTObject()`